src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -1062,6 +1095,305 @@ public <T> T[] toArray(T[] a) {
         }
     }
 
+    @jdk.internal.ValueBased
+    static final class ImmutableRegularEnumSet<E extends Enum<E>> extends AbstractImmutableSet<E>
+            implements RegularEnumSetCompatible<E>, Serializable {
+        @Stable
+        final long elements;
+
+        @Stable
+        final Class<E> elementType;
+
+        ImmutableRegularEnumSet(long elements, Class<E> elementType) {
+            this.elements = elements;
+            this.elementType = elementType;
+        }
+
+        @Override
+        public long elements() {
+            return elements;
+        }
+
+        @Override
+        public Class<E> elementType() {
+            return elementType;
+        }
+
+        @Override
+        public Iterator<E> iterator() {
+            return new ImmutableRESIterator();
+        }
+
+        private final class ImmutableRESIterator implements Iterator<E> {
+            long unseen = elements;
+            long lastReturned = 0;
+            final Enum<?>[] universe
+                = SharedSecrets.getJavaLangAccess().getEnumConstantsShared(elementType);
```

Comment:
I think the convention is to put `SharedSecrets.getJavaLangAccess()` instance in a static final field like `JLA` than calling it every time